packages/flutter/lib/src/widgets/text_selection.dart

```diff
@@ -0,0 +1,223 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/rendering.dart';
+
+import 'basic.dart';
+import 'framework.dart';
+import 'gesture_detector.dart';
+import 'overlay.dart';
+
+// TODO(mpcomplete): Need one for [collapsed].
+/// Which type of selection handle to be displayed. With mixed-direction text,
+/// both handles may be the same type. Examples:
+/// LTR text: 'the <quick brown> fox'
+///   The '<' is drawn with the [left] type, the '>' with the [right]
+/// RTL text: 'xof <nworb kciuq> eht'
+///   Same as above.
+/// mixed text: '<the nwor<b quick fox'
+///   Here 'the b' is selected, but 'brown' is RTL. Both are drawn with the
+///   [left] type.
+enum TextSelectionHandleType { left, right }
+
+/// Builds a handle of the given type.
+typedef Widget TextSelectionHandleBuilder(BuildContext context, TextSelectionHandleType type);
+
+/// The text position that a give selection handle manipulates. Dragging the
+/// [start] handle always moves the [start]/[baseOffset] of the selection.
+enum _TextSelectionHandlePosition { start, end }
+
+/// Manages a pair of text selection handles to be shown in an Overlay
+/// containing the owning widget.
+class TextSelectionHandles {
+  TextSelectionHandles({
+    this.selection,
+    this.renderObject,
+    this.onSelectionHandleChanged,
+    this.builder
+  });
+
+  /// A map of a _handles index to Position.
+  static final List<_TextSelectionHandlePosition> _positions =
+    <_TextSelectionHandlePosition>[
+      _TextSelectionHandlePosition.start,
+      _TextSelectionHandlePosition.end,
+    ];
+
+  // TODO(mpcomplete): what if the renderObject is removed or replaced, or
+  // moves? Not sure what cases I need to handle, or how to handle them.
+  final RenderEditableLine renderObject;
+  final ValueChanged<TextSelection> onSelectionHandleChanged;
+  final TextSelectionHandleBuilder builder;
+  TextSelection selection;
```

Comment:
Do you mean for the setter for this value to be public?  I would have expected mutable values to be private.